src/main/java/Duke.java

```diff
@@ -1,10 +1,291 @@
+                    for (int i = 1; i < parts.length; ++i) {
+                        if (!first) {
+                            taskDescription += " " + parts[i];
+                        } else {
+                            taskDescription += parts[i];
+                        }
+                        first = false;
+                    }
+                    Todo todo = new Todo(taskDescription);
+                    store.add(todo);
+                    System.out.println("     " + todo);
+                    System.out.println("     Now you have " + store.size() + " tasks in the list.");
+                } else if (parts[0].equals("deadline")) {
+                    if (parts.length == 1) {
+                        throw new InsufficientArgumentsException("     ☹ OOPS!!! The " +
+                                "description of a deadline cannot be empty.");
+                    }
+                    System.out.println("     Got it. I've added this task:");
+                    boolean flag = false, first = true;
+                    String dueDate = "";
+                    for (String s : parts) {
+                        if (flag) {
+                            if (!first) {
+                                dueDate += ' ' + s;
+                            } else {
+                                dueDate += s;
+                            }
+                            first = false;
+                        }
+                        if (s.equals("/by")) {
+                            flag = true;
+                        }
+                    }
+                    LocalDate due = LocalDate.parse(dueDate);
+                    String taskDescription = "";
+                    first = true;
+                    for (int i = 1; i < parts.length; ++i) {
+                        if (!parts[i].equals("/by")) {
+                            if (!first) {
+                                taskDescription += " " + parts[i];
+                            } else {
+                                taskDescription += parts[i];
+                            }
+                            first = false;
+                        } else {
+                            break;
+                        }
+                    }
+                    Deadline deadline = new Deadline(taskDescription, due);
+                    store.add(deadline);
+                    System.out.println("     " + deadline);
+                    System.out.println("     Now you have " + store.size() + " tasks in the list.");
+                } else if (parts[0].equals("event")) {
+                    if (parts.length == 1) {
+                        throw new InsufficientArgumentsException("     ☹ OOPS!!! The " +
+                                "description of an event cannot be empty.");
+                    }
+                    System.out.println("     Got it. I've added this task:");
+                    boolean flag = false, first = true;
```

Comment:
Perhaps a more descriptive name for the boolean variable (instead of "flag") will be better?